src/lib/Microsoft.Health.Events/Telemetry/Exceptions/EventHubExceptionTelemetryProcessor.cs

```diff
@@ -68,24 +72,40 @@ public static (Exception customException, string errorName) CustomizeException(E
                     return (new InvalidEventHubException(message, exception, errorName), errorName);
 
                 case InvalidOperationException _:
-                    message = "Verify that the provided Event Hub contains the provided consumer group.";
+                    message = EventResources.EventHubInvalidConsumerGroup;
                     errorName = nameof(EventHubErrorCode.ConfigurationError);
                     return (new InvalidEventHubException(message, exception, errorName), errorName);
 
+                case MsalServiceException _:
+                    var msalErrorCode = ((MsalServiceException)exception).ErrorCode;
+                    message = EventResources.ManagedIdentityAuthenticationError;
+                    return (new ManagedIdentityAuthenticationError(message, exception, msalErrorCode), msalErrorCode);
+
+                case RequestFailedException _:
+                    var errorCode = ((RequestFailedException)exception).ErrorCode;
+
+                    if (string.Equals(errorCode, EventResources.SecretNotFound, StringComparison.CurrentCultureIgnoreCase) || message.Contains(EventResources.SecretNotFound, StringComparison.CurrentCultureIgnoreCase))
+                    {
+                        message = EventResources.ManagedIdentityCredentialNotFound;
```

Comment:
If we get here is it always guaranteed that we have an issue with Managed Identity? Looking at the code I think that we also support [ConnectionString](https://microsofthealth.visualstudio.com/Health/_git/workspace-platform?path=/iot-connector/core/iomt-fhir/Microsoft.Health.Fhir.Ingest.Console.PaaS/EventHubs/EventHubConsumerClientFactory.cs&version=GBmain&line=32&lineEnd=33&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents).

Reply:
If using the ConnectionString auth type, since there is no secret involved that we fetch from the key vault, I think only in the custom auth type, we use the credential bundle and can expect to get this error if it doesn't exist in the keyvault.